### PR TITLE
[move-only] Initial patch for move only address checking.

### PR DESCRIPTION
**Pull request by gottesmm**
I handle now var, inout, mutating self that are non-trivial/loadable. Also, I handle inits correctly. Still a work in progress, but enough is done to get something into tree.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

File: include/swift/SIL/MemAccessUtils.h [hunk 1]
```diff
@@ -1536,6 +1536,8 @@ inline Operand *getAccessProjectionOperand(SingleValueInstruction *svi) {
   // would not be casts, but will likely be eliminated with opaque values.
   case SILInstructionKind::OpenExistentialAddrInst:
   case SILInstructionKind::UncheckedTakeEnumDataAddrInst:
+  // We look through mark must check.
+  case SILInstructionKind::MarkMustCheckInst:
```
Comment (eeckstein) [inline include/swift/SIL/MemAccessUtils.h:1540]: Please also add this to `AddressUseDefWalker.walkUpDefault` and `AddressDefUseWalker.walkDownDefault` in `WalkUtils.swift` (and a test in `SILOptimizer/accessutils.sil`)

**Review by eeckstein — DISMISSED** [review]

File: include/swift/SIL/MemAccessUtils.h (hunk 1 above)
Comment (gottesmm) [reply include/swift/SIL/MemAccessUtils.h:1540]: @eeckstein https://github.com/apple/swift/pull/60989

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
One thing to keep in mind with this patch series is that I left out the address only tests that I have on a branch. This was done on purpose since until I fix SILGen to make it so that we borrow move only var like things when we pass them to function arguments, we hit a ton of spurious errors on the tests. I am going to upload those tests together with the borrow fix just to save my hands a ton of unnecessary typing.

**Pull request merged**